### PR TITLE
[air][data] move train_test_split to ray.data.Dataset

### PR DESCRIPTION
**Pull request by matthewdeng**
Signed-off-by: Matthew Deng <matt@anyscale.com>

## Why are these changes needed?

Moving `ray.air.train_test_split` to `ray.data.Dataset.train_test_split`.

## Related issue number

Closes #https://github.com/ray-project/ray/issues/27024

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — COMMENTED** [review]
Do we just want to move it into dataset? We already have a bunch of other ML ops there so may as well at this point. It's weird to have it standalone outside.

```
dataset.train_test_split()
```

**Review by richardliaw — APPROVED** [review]
nice!

**Review by clarkzinzow — APPROVED** [review]

**Comment by zhe-thoughts** [comment]
We need to update API references right? https://docs.ray.io/en/master/ray-air/package-ref.html

**Comment by richardliaw** [comment]
Seems fine Zhe, updates automatically

File: python/ray/data/dataset.py [hunk 1]
```diff
@@ -1171,6 +1171,63 @@ def split_proportionately(self, proportions: List[float]) -> List["Dataset[T]"]:
 
         return self.split_at_indices(split_indices)
 
+    def train_test_split(
+        self,
+        test_size: Union[int, float],
+        *,
+        shuffle: bool = False,
```
Comment (jianoaix) [inline python/ray/data/dataset.py:1178]: Shuffle is a transformation while splitting is a consumption. Can this be dropped and let user do `ds.shuffle().train_test_split()`?
